mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionAvailbilityAwareAcivity.java

```diff
@@ -0,0 +1,148 @@
+package org.openhab.habdroid.core.connection;
+
+import android.app.Fragment;
+import android.content.Intent;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.support.v4.app.TaskStackBuilder;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.openhab.habdroid.R;
+import org.openhab.habdroid.core.connection.exception.ConnectionException;
+import org.openhab.habdroid.core.connection.exception.NetworkNotAvailableException;
+import org.openhab.habdroid.core.connection.exception.NetworkNotSupportedException;
+import org.openhab.habdroid.core.connection.exception.NoUrlInformationException;
+import org.openhab.habdroid.ui.OpenHABMainActivity;
+import org.openhab.habdroid.ui.OpenHABPreferencesActivity;
+
+import static org.openhab.habdroid.ui.OpenHABPreferencesActivity.NO_URL_INFO_EXCEPTION_EXTRA;
+import static org.openhab.habdroid.ui.OpenHABPreferencesActivity.NO_URL_INFO_EXCEPTION_MESSAGE;
+
+public abstract class ConnectionAvailbilityAwareAcivity extends AppCompatActivity {
```

Comment:
This thing should be named ConnectionAvailabilityAwareActivity.

Reply:
Done!